### DISCUSSION TITLE
Invalidating routes

### DISCUSSION THREAD

**Discussion by s-cork** [TanStack/router#411] [Q&A] [answered]
Situation:
I have a route that has a `loader` with a `loaderMaxAge`
I would like to invalidate that route when I know the data is about to be stale.
say it's a list of posts and I am about to delete a post.

Perhaps I'd do it via navigate:
pseudo code:
```js
const onPostDelete = () => {
    router.navigate({to: "/posts", replace: true, invalidate: true});
}
```

Obviously that's not valid API but I can't work out whether the API for this exists?

I see there is an API for `router.invalidateRoute({to: "/posts"})`
but that only seems to invalidate matches where they exist in the current `state.matches`
however the route I want to invalidate is not part of `state.matches`

**Comment by tannerlinsley** [comment, marked as answer]
It should invalidate all. Bug!
